src/main/java/javaslang/collection/Stream.java

```diff
@@ -148,6 +148,9 @@ public boolean hasNext() {
 
             @Override
             public T next() {
+                if (i >= elements.length) {
+                    throw new NoSuchElementException();
+                }
                 return elements[i++];
```

Comment:
In general you are of course right. Here, this check is unnecessary because the Iterator is internally used only. The `Stream.ofAll(Iterator)` uses a for-loop to iterate the elements. Please remove the check to save cycles.